### DISCUSSION TITLE
Sidebar not showing items

### DISCUSSION THREAD

**Discussion by CosminCAS123** [vikramlearning/blazorbootstrap#939] [Q&A] [answered]
Hey! New to blazor bootstrap and blazor in general(Web App interactive auto template ). In client project, I defined a layout that'll use the SideBar component . I simplified it for this issue ( no data provider) :
```
@inherits LayoutComponentBase


<div class="bb-page">
    <Sidebar Title="For relationships" IconName="IconName.Apple">
                 < NavItem Href="/home" IconName="IconName.House" Text="Home" />
    
    </Sidebar>

    <main class="content">
        @Body
    </main>
</div>

```

then the Home page (client project) looks like this : 
```
@page "/home"
@rendermode InteractiveAuto
@layout Layouts.SidebarLayout

<h1>HOME PAGE WITH SIDEBAR LAYOUT</h1>

```

And the result is : 
![image](https://github.com/user-attachments/assets/67ad4378-3749-4d63-98f8-dbea54778828)

I also tried the layout on my starting  auth interactive SSR pages (login and register) from the server, and the same thing happens. 

I installed blazor bootstrap accordingly to the web app interactive auto tutorial. If you need any more information ,please tell me. Thanks!

**Comment by gvreddy04** [comment]
@CosminCAS123 Thank you for trying BlazorBootstrap. Here is the example:

```cshtml
<Sidebar Href="/"
         IconName="IconName.BootstrapFill"
         Title="Blazor Bootstrap"
         DataProvider="SidebarDataProvider" />

@code {
    IEnumerable<NavItem>? navItems;

    private async Task<SidebarDataProviderResult> SidebarDataProvider(SidebarDataProviderRequest request)
    {
        if (navItems is null)
            navItems = GetNavItems();

        return await Task.FromResult(request.ApplyTo(navItems));
    }

    private IEnumerable<NavItem> GetNavItems()
    {
        navItems = new List<NavItem>
        {
            new NavItem { Href = "/getting-started", IconName = IconName.HouseDoorFill, Text = "Getting Started"},
            new NavItem { Href = "/alerts", IconName = IconName.CheckCircleFill, Text = "Alerts"},
            new NavItem { Href = "/autocomplete", IconName = IconName.InputCursorText, Text = "Auto Complete"},
            new NavItem { Href = "/breadcrumb", IconName = IconName.SegmentedNav, Text = "Breadcrumb"},
            new NavItem { Href = "/sidebar", IconName = IconName.LayoutSidebarInset, Text = "Sidebar"},
        };
        return navItems;
    }
}
```
![image](https://github.com/user-attachments/assets/25d0aa5c-2882-4738-a71e-f505cd84cfac)

Link: https://demos.blazorbootstrap.com/sidebar

**Reply by CosminCAS123** [reply]
yes, i followed the tutorial , but I simplified my sidebar for the sake of example. still endless loading happening.

**Reply by zabeda** [reply]
I am having a similar problem and I think I may have tracked down what is happening. It appears that the menu has the collapse class added to it, so it is never displayed. If I manually remove the collapse class in the browser, the menu appears. Is there something else that needs to be done to have this not include the collapse class? I looked at the code for SideBar and I didn't see how this could be done from the outside. It looks like collapseNavMenu is initialized to true, but there isn't a way to flip it.

Any assistance would be appreciated. Thank you.

![image](https://github.com/user-attachments/assets/ec5977dd-152c-49b1-8bc1-f8f87336cb76)

**Reply by Soundman32** [reply]
I'm seeing the same thing.  Is there any fix for this?

**Reply by zabeda** [reply]
I ended up shelving the design that needed the sidebar, so stopped working on this. I had started looking at creating a custom component that extended BlazorBootstrapComponentBase for the sidebar. I think it was using the code from Sidebar in the project. I checked and I don't have the code. Sorry can't help more.

**Reply by EldritchMuppet** [reply]
Your problem is you dont have it in interactiveserver rendering mode so it's not calling the dataprovider

You can set the mode in a bunch of places but the one used in the blazorbootstrap getting started doc is to use in your app component;
`<Routes @rendermode=@InteractiveServer />`

**Reply by jtaylor1098** [reply]
Hey, everyone,
I just started working on a project, and I am having the same issue with the collapse class attached to the "bb-sidebar-content" within the sidebar component. I've attempted to use display: inherit; within the class and wrapping in a div. That works for desktop, but it hinders a responsive design (mobile). In "Sidebar2" they use a "Task.Delay(2000);" so I wouldn't be surprised if that's their solution. It works, but also a temporary solution. Considering that there still doesn't seem to be an answer, I think I'll move on to a different UI library, but I'm hoping someone will respond with a solution.

Sidebar2: Notice the "collapse" class on "bb-sidebar2-content". If you use the toggle button under 640px width, it'll show when reverting to desktop.
<img width="916" height="423" alt="Screenshot 2026-02-18 154511" src="https://github.com/user-attachments/assets/29010373-a20b-4ce0-82cc-eb8933cd1b48" />

**Reply by s8jt8zw8jf-lang** [reply]
Yes, I'm seeing the collapse as well, and the nav-items are not being displayed.
.net 10 project from visual studio (2026) temple for blazor server app.
Using blazor 3.5.0.  Tried switching back to 3.4, same.

Followed instructions for installing blazor, server, .net 8.(https://docs.blazorbootstrap.com/getting-started/blazor-webapp-server-global-net-8)

any suggestions welcome please....

**Reply by gvreddy04** [reply]
> Yes, I'm seeing the collapse as well, and the nav-items are not being displayed. .net 10 project from visual studio (2026) temple for blazor server app. Using blazor 3.5.0. Tried switching back to 3.4, same.
> 
> Followed instructions for installing blazor, server, .net 8.(https://docs.blazorbootstrap.com/getting-started/blazor-webapp-server-global-net-8)
> 
> any suggestions welcome please....

Thank you for sharing your feedback. Please create a sample GitHub repo to reproduce the issue. I'll take a look. Do not share .zip files.

**Reply by s8jt8zw8jf-lang** [reply, marked as answer]
My bad.
Set the render to be per page instead of global.
Tried to edit the project to global location, but that failed to resolve the issue.
Started a fresh new project using global location, and good to go now.
Seems for layout page you cannot define a render location, so has got to be global if you need to do any rendering in these files.

now to migrate work done to new project..
